### PR TITLE
refactoring test_subjob.py

### PR DESCRIPTION
**Pull request by ethomas2**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @ethomas2 has signed the CLA. Thanks for the pull request!

**Comment by ethomas2** [comment]
@josephharrington made the changes you suggested. Since this commit really belongs with the last one, is there a proper way to squash the two that doesn't involve a git push -f to master? The -f seems a little heavy handed

**Comment by tjlee0909** [comment]
-f is what I usually do.

File: test/unit/master/test_subjob.py [hunk 1]
```diff
@@ -36,9 +36,9 @@ def setUp(self):
             ],
         )
 
-    def test_set_atoms_subjob_id(self):
+    def test_subjob_constructor_sets_subjob_id_on_atoms(self):
         atoms = [Mock(), Mock()]
-        self._subjob._set_atoms_subjob_id(atoms, 4)
+        Subjob(1, 4, Mock(), Mock(), atoms)
```
Comment (aptxkid) [inline test/unit/master/test_subjob.py]: How about using keyword arguments? It makes it much more readable.

**Comment by aptxkid** [comment]
Push -f to master is disruptive to people already have the latest change. I would suggest just merge this PR separately.

File: test/unit/master/test_subjob.py (hunk 1 above)
Comment (ethomas2) [inline test/unit/master/test_subjob.py]: whoa, didn't even know you could use keyword arguments unless the function declaration looked like
def f(a=foo, b=bar)

**Comment by josephharrington** [comment]
+1 to just merging separately.

LGTM :+1: :+1: :+1: Thanks for doing this!

**Comment by aptxkid** [comment]
:+1:

**Pull request merged**